xdsl/dialects/ptr.py

```diff
@@ -43,6 +53,9 @@ class PtrAddOp(IRDLOperation):
 
     assembly_format = "$addr `,` $offset attr-dict `:` `(` type($addr) `,` type($offset) `)` `->` type($result)"
 
+    def __init__(self, addr: SSAValue, offset: SSAValue):
```

Comment:
```suggestion
    def __init__(self, addr: SSAValue | Operation, offset: SSAValue | Operation):
```
Are we moving away from this?

Reply:
I never liked it, so it's my current quiet rebellion. Note that now that we have generic SSA values, we can actually do things like specify the types of values, which is impossible if we also allow Operation.

Reply:
fair